### PR TITLE
Bruce/upgrade repo

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -10,43 +10,43 @@
     "resolveJsonModule": true,
     "esModuleInterop": true,
     "module": "esNext",
-    "target": "es5",
+    "target": "es6",
```
Comment (phpgeekfr) [inline tsconfig.base.json:13]: I don't know the possible side effects of this

File: libs/testing/cypress/browser/src/lib/festival.ts [hunk 2]
```diff
@@ -136,6 +136,7 @@ export function goToInvitations() {
 export function refuseInvitationScreening() {
   cy.get('invitation-view [test-id=more]').first().click();
   cy.get('[test-id=decline-invitation]').click();
+  cy.wait(5000);
```
Comment (phpgeekfr) [inline libs/testing/cypress/browser/src/lib/festival.ts:139]: without wait, request to DB was not ended (worked fine before)

File: apps/backend-functions/src/data/internals.ts [hunk 3]
```diff
@@ -39,9 +40,7 @@ export function createPublicUserDocument(user: Partial<PublicUser> = {}) {
   }
 }
 
-export type Timestamp = admin.firestore.Timestamp;
-
-export function createDocumentMeta(meta: Partial<DocumentMeta<Timestamp>> = {}): DocumentMeta<Timestamp> {
+export function createDocumentMeta(meta: Partial<DocumentMeta<any>> = {}): DocumentMeta<any> { // TODO #7273 #8006 any
```
Comment (phpgeekfr) [inline apps/backend-functions/src/data/internals.ts:43]: I had to put any to fix Timestamp errors between front and back

File: apps/festival/festival/tsconfig.app.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "outDir": "../../../dist/out-tsc",
-    "types": []
+    "types": ["offscreencanvas"]
```
Comment (phpgeekfr) [inline apps/festival/festival/tsconfig.app.json:5]: required for twilio

File: libs/firebase-utils/src/lib/anonymize/lib.spec.ts [hunk 5]
```diff
@@ -211,7 +211,7 @@ describe('Test _META anonymization function', () => {
     };
 
     const doc = anonymizeDocument(titleRecord);
-    expect(doc.content.startedAt._seconds).toBeDefined();
+    expect(doc.content.startedAt.seconds).toBeDefined();
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/anonymize/lib.spec.ts:214]: new Timestamp object

File: apps/financiers/financiers/jest.config.js [hunk 6]
```diff
@@ -11,6 +11,7 @@
   transform: {
     '^.+\\.(ts|html)$': 'ts-jest',
   },
+  transformIgnorePatterns: ['node_modules/(?!.*.mjs$)'],
```
Comment (phpgeekfr) [inline apps/financiers/financiers/jest.config.js:14]: @george43g  I had to add this on some` jest.config.js`

File: libs/consents/jest.config.js [hunk 7]
```diff
@@ -1,13 +1,17 @@
...
   globals: {
     'ts-jest': {
       stringifyContentPathRegex: '\\.(html|svg)$',
       tsconfig: '<rootDir>/tsconfig.spec.json',
     },
   },
+  transform: {
+    '^.+.(ts|mjs|js|html)$': 'jest-preset-angular'
+  },
+  testEnvironment: '<rootDir>/jest-environment.js',
```
Comment (phpgeekfr) [inline libs/consents/jest.config.js:13]: @george43g  needed for unit testing front services & using firebase functions. see `libs/consents/jest-environment.js`

File: migrations.json [hunk 8]
```diff
@@ -0,0 +1,362 @@
+{
```
Comment (phpgeekfr) [inline migrations.json:1]: @george43g does this file needs to be committed ?

File: apps/backend-functions/src/data/internals.ts (hunk 3 above)
Comment (george43g) [reply apps/backend-functions/src/data/internals.ts:43]: I'll see if there's a cleaner way to do this

File: libs/movie/src/lib/movie/+state/movie.service.ts [hunk 9]
```diff
@@ -131,6 +137,6 @@ export class MovieService extends CollectionService<MovieState> {
           status,
         },
       },
-    }));
+    } as any)); // TODO #7273
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/+state/movie.service.ts:140]: `as any` was required (not needed before)

File: libs/movie/src/lib/movie/dashboard/actions/actions.component.ts [hunk 10]
```diff
@@ -56,7 +56,7 @@ export class DashboardActionsShellComponent {
                 access: false,
               },
             },
-          }));
+          } as any)); // TODO #7273
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/dashboard/actions/actions.component.ts:59]: `as any` was required (not needed before)

**Comment by phpgeekfr** [comment]
@george43g when running `npm start emulators` :
```
⚠  Your requested "node" version "12" doesn't match your global version "16"
```

where can this be changed ?

File: apps/backend-functions/src/data/internals.ts (hunk 3 above)
Comment (phpgeekfr) [reply apps/backend-functions/src/data/internals.ts:43]: see #8006

File: libs/utils/src/lib/maintenance/index.ts [hunk 11]
```diff
@@ -35,8 +34,8 @@ export function _isInMaintenance(maintenanceDoc: IMaintenanceDoc, delay = EIGHT_
       // Wait `delay` minutes before allowing any operation on the db.
       // this prevents triggering firebase events.
       // NOTE: this is hack-ish but good enough for our needs! we'll revisit this later.
-      const now = firebase.firestore.Timestamp.now();
-      return maintenanceDoc.endedAt.toMillis() + delay > now.toMillis();
+      const now = Date.now();
```
Comment (phpgeekfr) [inline libs/utils/src/lib/maintenance/index.ts:37]: like this, both back and front can use this function #8006

File: package.json [hunk 12]
```diff
@@ -2,6 +2,10 @@
   "name": "blockframes",
   "version": "0.0.2",
   "license": "MIT",
+  "engines": {
+    "node": "16.14.2",
```
Comment (phpgeekfr) [inline package.json:6]: @GrandSchtroumpf updated to 16.14.2 as it is the LTS version (not 16.14.0)

**Comment by phpgeekfr** [comment]
> @george43g when running `npm start emulators` :
> 
> ```
> ⚠  Your requested "node" version "12" doesn't match your global version "16"
> ```
> 
> where can this be changed ?

solved by updating engine in backend-functions package.json

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**